source/beamline/AMBeamlineControlGroupStatus.cpp

```diff
@@ -101,32 +101,17 @@ QList<AMBeamlineControl*> AMBeamlineControlGroupStatus::goodControls() const
 
 bool AMBeamlineControlGroupStatus::addControl(AMBeamlineControl *control)
 {
-	bool result = false;
-
-	if (control && !hasChildControl(control)) {
-		addChildControl(control);
-		result = true;
-	}
-
-	return result;
+        return addChildControl(control);
 }
 
 bool AMBeamlineControlGroupStatus::removeControl(AMBeamlineControl *control)
 {
-	bool result = false;
-
-	if (control && hasChildControl(control)) {
-		removeChildControl(control);
-		result = true;
-	}
-
-	return result;
+        return removeChildControl(control);
 }
 
 bool AMBeamlineControlGroupStatus::clearControls()
 {
-	clearChildControls();
-	return true;
+        return clearChildControls();
 }
```

Comment:
I wonder if either of these three functions is really needed any more. I guess they restrict adding controls (through this function at least) to the AMBeamlineControl type (and sub-types), but otherwise they're just an alias for addChildControl, removeChildControl and clearChildControls(). @helfrij will have more of an idea